src/Sarif/Visitors/RunMergingVisitor.cs

```diff
@@ -1,82 +1,113 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.Collections.Generic;
 using System.Diagnostics;
 
 namespace Microsoft.CodeAnalysis.Sarif.Visitors
 {
     /// <summary>
-    /// This class is used to update indices for data that is merged for various
-    /// reasons, including result matching operations.
+    ///  RunMergingVisitor merges Results from multiple Runs. 
+    ///  It gathers things referenced by the Results, remaps indices in the Results to the merged collections,
+    ///  and then populates the desired run with the merged sets.
+    ///  
+    ///  Used by Baselining and Run.MergeResultFrom(Run).
     /// </summary>
-    public class RemapIndicesVisitor : SarifRewritingVisitor
+    /// <remarks>
+    ///  Usage:
+    ///   var visitor = new RunMergingVisitor();
+    ///   
+    ///   visitor.VisitRun(primary);
+    ///   visitor.VisitRun(additional);
+    ///   
+    ///   visitor.PopulateMergedRun(primary);
+    /// </remarks>
+    public class RunMergingVisitor : SarifRewritingVisitor
     {
-        public RemapIndicesVisitor(
-            IList<Artifact> currentArtifacts,
-            IList<LogicalLocation> currentLogicalLocations)
-        {
-            CurrentArtifacts = new List<Artifact>();
-            CurrentLogicalLocations = new List<LogicalLocation>();
+        private List<Result> Results { get; set; }
```

Comment:
>Results [](start = 29, length = 7)

It looks like these first four properties can be `{ get; }`-only.

Reply:
Fixed.